frame/support/src/traits.rs

```diff
@@ -2035,6 +2065,17 @@ pub trait Hooks<BlockNumber> {
 	/// The block is being finalized. Implement to have something happen.
 	fn on_finalize(_n: BlockNumber) {}
 
+	/// This will be run when the block is being finalized (before on finalize).
+	/// Implement to have something happen using the remaining weight.
+	/// Returns the weight used, passes it to the next on idle hook if exists.
+	/// Will not fire if the remaining weight is below 0.
```

Comment:
this isn't currently true, is it?

Reply:
it is if the weight is zero before the first on idle hook, not if the weight gets to zero by an on idle hook https://github.com/paritytech/substrate/pull/8209/files#diff-71647e596916530d84fd02546feb833ff703dbf19397d076c499ab7dbf265da2R376